lib/broadway/options.ex

```diff
@@ -218,10 +218,27 @@ defmodule Broadway.Options do
               """
             ],
             batch_size: [
-              type: :pos_integer,
+              type: {:or, [:pos_integer, {:fun, 2}]},
```

Comment:
I would prefer to make it a tuple `{fun, acc}`. I think a clearer API is better than passing `nil` as default. You can always encapsulate it using a module or function, something like:

    batch_size: MyApp.byte_batcher()

WDYT?

Reply:
Originally, I also thought this structure is ok.  But it seems hard to define the Nimbler options for either positive integer or a tuple of function with an accumulator number.

Reply:
You will need to define a custom validation function yeah!